src/Compilers/Core/Portable/SourceGeneration/Nodes/CombineNode.cs

```diff
@@ -44,29 +52,50 @@ public NodeStateTable<(TInput1, TInput2)> UpdateStateTable(DriverStateTable.Buil
 
             // get the input2 item
             var isInput2Cached = input2Table.IsCached;
-            TInput2 input2 = input2Table.Single();
+            (TInput2 input2, IncrementalGeneratorRunStep? input2Step) = input2Table.Single();
 
             // append the input2 item to each item in input1 
             foreach (var entry1 in input1Table)
             {
-                var state = (entry1.state, isInput2Cached) switch
+                var stopwatch = SharedStopwatch.StartNew();
```

Comment:
>stopwatch [](http://example.com/codeflow?start=20&length=9)

What are we timing? The time it takes to get `entry1.Item`?

Reply:
We're basically timing how long it takes to create the new value, aka create the ValueTuple instance with the two items. If we don't think this is worthwhile (I'm dubious of its value) we can change the CombineNode execution time to always be `TimeSpan.Zero`.